### PR TITLE
[Merged by Bors] - Add tag-and-build command to makefile

### PR DESCRIPTION
**Pull request by noamnelke**
## Motivation
Enable less error prone tagging and building of a new version.

## Changes
Add `tag-and-build` and `list-versions` commands to `Makefile`.

## Usage
```
make list-versions
make tag-and-build VERSION=v0.1.11
``` 

Also added version release instructions to the wiki: 
https://github.com/spacemeshos/go-spacemesh/wiki/Version-Release-Instructions

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -161,6 +161,24 @@ cover:
 .PHONY: cover
 
 
+tag-and-build:
+	@git diff --quiet || (echo "working directory must be clean"; exit 2)
+	echo ${VERSION} > version.txt
```
Comment (y0sher) [inline Makefile:166]: isn't `VERSION` being read from the `version.txt` file why write it back?

Comment (noamnelke) [reply Makefile:166]: See what I wrote about usage. The idea is to overwrite this when calling to change the version number. This is tested and works.

If you have a nicer suggestion, I’m open to it...

Comment (y0sher) [reply Makefile:166]: so what you pass in the command overrides the first line which reads from file ?
I currently don't have a better suggestion :)

**Review by y0sher — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (noamnelke) [reply Makefile:166]: yes, based on [this answer in SO](https://stackoverflow.com/a/2826178/1225467):

>**From command line** - `make` can take variable assignments as part of his command line, mingled with targets:
>
>```
>make target FOO=bar
>```
>But then all assignments to `FOO` variable within the makefile will be ignored unless you use the override directive in assignment.

(tested and it works)

**Comment by noamnelke** [comment]
bors merge

**Pull request closed without merging**